test/VaultRegistry.test.ts

```diff
@@ -1,445 +1,943 @@
+                ).to.equal(this.nft);
+            });
+
+            describe("access control:", () => {
+                it("allowed: any address", async () => {
+                    await withSigner(randomAddress(), async (s) => {
+                        await expect(
+                            this.subject
+                                .connect(s)
+                                .nftForVault(this.erc165Mock.address)
+                        ).to.not.be.reverted;
+                    });
+                });
+            });
+
+            describe("edge cases", () => {
+                describe("when Vault is not registered in VaultRegistry", () => {
+                    it("returns zero", async () => {
+                        expect(
+                            await this.subject.nftForVault(randomAddress())
+                        ).to.equal(0);
+                    });
+                });
+            });
+        });
+
+        describe("#isLocked", () => {
+            it("checks if token is locked (not transferable)", async () => {
+                await this.subject.connect(this.ownerSigner).lockNft(this.nft);
+                expect(await this.subject.isLocked(this.nft)).to.be.true;
+            });
+
+            describe("access control:", () => {
+                it("allowed: any address", async () => {
+                    await withSigner(randomAddress(), async (s) => {
+                        await expect(this.subject.connect(s).isLocked(this.nft))
+                            .to.not.be.reverted;
+                    });
+                });
+            });
+
+            describe("edge cases", () => {
+                describe("when VaultRegistry NFT is not registered in VaultRegistry", () => {
+                    it("returns false", async () => {
+                        expect(await this.subject.isLocked(this.randomNft)).to
+                            .be.false;
+                    });
+                });
+            });
+        });
+
+        describe("#protocolGovernance", () => {
+            it("returns ProtocolGovernance address", async () => {
+                expect(await this.subject.protocolGovernance()).to.eq(
+                    this.protocolGovernance.address
+                );
+            });
+
+            describe("access control:", () => {
+                it("allowed: any address", async () => {
+                    await withSigner(randomAddress(), async (s) => {
+                        await expect(this.subject.protocolGovernance()).to.not
+                            .be.reverted;
+                    });
+                });
+            });
+        });
+
+        describe("#stagedProtocolGovernance", () => {
+            it("returns ProtocolGovernance address staged for commit", async () => {
+                await this.subject
+                    .connect(this.admin)
+                    .stageProtocolGovernance(
+                        this.newProtocolGovernance.address
```

Comment:
can we replace it with randomAddress() ?

Reply:
with current functionality it will work